docs/table-migration.md

```diff
@@ -0,0 +1,129 @@
+---
+title: "Table Migration"
+url: table-migration
+weight: 400
+menu: main
+---
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+# Hive and Spark Table Migration
+Iceberg supports converting Hive and Spark tables to Iceberg tables through Spark Procedures. For specific instructions on how to use the procedures, please refer to the [Table Migration via Spark Procedures](../spark-procedures/#table-migration) page.
```

Comment:
Currently, the [Spark Procedures](https://iceberg.apache.org/docs/latest/spark-procedures/#table-migration) contains a comprehensive explanation on migrating from Hive or Spark to Iceberg. So I made a pointer to that page here. I think another option would be to move everything to this newly added page and leave a pointer at [Spark Procedures](https://iceberg.apache.org/docs/latest/spark-procedures/#table-migration) which point to the new migraiton page.

More thoughts are welcomed on the design. Please feel free to share your opinions on this. Thank you in advance.

Reply:
Yes I agree we should point to the Spark procedure instead of repeat the information. I am expecting this dedicated page to explain more about how the migration works, including (1) how snapshot, migrate are different, (2) how to use snapshot/migrate + add files to continuously migrate a table, (3) how the migration works differently for Hive, Delta and in the future Hudi. 

We can add some graphs and explain each case, and then point to the Spark procedure for actual usage of it. For Delta, we can add a section in java API documentation and point to that instead of a Spark procedure.